<launch>

<arg name="headless" default="false" />
<arg name="separate_pose" default="false" />
<arg name="init_mode" default="both" />

<arg name="debug" default="false" />
<arg name="debug_prefix" value="gdb --args" if="$(arg debug)"/>
<arg name="debug_prefix" value="" unless="$(arg debug)"/>

<!-- mcptam node -->
<node name="mcptam" pkg="mcptam" type="mcptam" clear_params="true" output="screen" launch-prefix="$(arg debug_prefix)">
  <rosparam command="load" file="$(find mcptam)/groups/$(arg group_name).yaml" />
  
  <remap from="reset" to="mcptam/reset" />
  <param name="image_transport" type="string" value="compressed"/>
  
  <param name="get_pose_separately" type="bool" value="$(arg separate_pose)" />   
  <!-- <param name="extrinsic_scale" type="double" value="0.95426" /> -->
  
  <param name="dynamic_sync" type="bool" value="true" />
  
  <!--
  <param name="camera_pose_file" type="string" value="$(find mcptam)/poses/poses_calibrator.dat" />
  -->
<!--
  <rosparam command="load" file="$(find mcptam)/masks/masks_1,2,3,4.yaml" />
  <param name="masks_dir" type="string" value="$(find mcptam)/masks" />
-->

  <param name="mm_init_point_mode" type="string" value="$(arg init_mode)" />
  <param name="mm_init_point_max_num" type="int" value="200" />

  <param name="kf_distance_mean_diff_fraction" type="double" value="0.1" /> 
  <param name="mm_max_scaled_mkf_dist" type="double" value="0.3" />
  <param name="mm_outlier_multiplier" type="double" value="1.1" />
  <param name="mm_init_cov_thresh" type="double" value="0.01" />
  <param name="kf_adaptive_thresh" type="bool" value="true" />
  <param name="level_zero_points" type="bool" value="true" />
  
  <param name="adjuster_recent_min_size" type="int" value="60" />
  
  <!-- For no GUI, set headless:=true -->
  <param name="headless" type="bool" value="$(arg headless)" />
</node>

<node pkg="tf" type="static_transform_publisher" name="world_tf" args="0 0 0 0 0 3.14159 world vision_world 100" />

</launch>

